concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/10 Gray Boxes Appear After Data Loading/25 Eliminate Side Effects.md

```diff
@@ -0,0 +1,103 @@
+
+    In such cases, save the scroll position on hide and restore it on show:
+
+        <!-- tab: app.component.ts -->
+        export class AppComponent {
+            // ...
+            onContentVisibilityChange(visible) {
+                if (!visible) {
+                    this.gridScrollPosition = this.dataGrid.instance.getScrollable().scrollTop();
+                } else if (this.gridScrollPosition > 0) {
+                    this.dataGrid.instance.getScrollable().scrollTo({ top: this.gridScrollPosition }); 
+            }
+        }
+    
+    Alternatively, you can use CSS to hide the component. In this case, call the [updateDimensions](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#updateDimensions) method.
+
+##### Vue
+
+- If you use conditional rendering to display DataGrid or TreeList, and virtual scrolling is enabled, the following happens:
+
+    - **on hide**    
+    The scroll position is discarded.
+
+    - **on show**    
+    The scroll position is set to the first page.
+
+    If DataGrid or TreeList was opened, for example, on the page 5 before it was hidden, the gray boxes are displayed after the component is shown for the second time. This happens because data source is loaded for the page 5 only.
+
+    In such cases, save the scroll position on hide and restore it on show:
+
+        <!-- tab: App.vue -->
+        <script>
+            export default {
+                // ...
+                methods: {
+                    onContentVisibilityChange(visible) {
+                        if (!visible) {
+                            this.gridScrollPosition = this.dataGrid.instance.getScrollable().scrollTop();
+                        } else if (this.gridScrollPosition > 0) {
+                            this.dataGrid.instance.getScrollable().scrollTo({ top: this.gridScrollPosition }); 
+                        }
+                    }
+                }
+            }
+        </script>
+    
+    Alternatively, you can use CSS to hide the component. In this case, call the [updateDimensions](https://js.devexpress.com/Documentation/ApiReference/UI_Components/dxDataGrid/Methods/#updateDimensions) method.
+
+##### React
+
+- In certain cases, row placeholders appear due to redundant re-rendering. To learn more about such issues, refer to the following guide: [Optimize Performance](/Documentation/Guide/React_Components/Optimize_Performance).
+
+- If you use conditional rendering to display DataGrid or TreeList, and virtual scrolling is enabled, the following happens:
+
+    - **on hide**    
+    The scroll position is discarded.
+
+    - **on show**    
+    The scroll position is set to the first page.
+
+    If DataGrid or TreeList was opened, for example, on the page 5 before it was hidden, the gray boxes are displayed after the component is shown for the second time. This happens because data source is loaded for the page 5 only.
```

Comment:
Likewise.

```suggestion
    If DataGrid or TreeList was opened, for example, on the page 5 before it was hidden, the gray boxes are displayed after the component is shown for the second time. This happens because the data source is loaded for the page 5 only.
```